server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/GroupMappingTest.java

```diff
@@ -0,0 +1,379 @@
+
+        webAdminGuiceProbe = jamesServer.getProbe(WebAdminGuiceProbe.class);
+
+        message = MimeMessageBuilder.mimeMessageBuilder()
+            .setSubject("test")
+            .setText(MESSAGE_CONTENT)
+            .build();
+    }
+
+    @After
+    public void tearDown() {
+        jamesServer.shutdown();
+    }
+
+    @Test
+    public void messageShouldRedirectToUserWhenBelongingToGroup() throws Exception {
+        RestAssured.given()
+            .port(webAdminGuiceProbe.getWebAdminPort())
+        .when()
+            .put(GroupsRoutes.ROOT_PATH + "/" + GROUP1_ADDRESS + "/" + USER_DOMAIN1)
+        .then()
+            .statusCode(201);
+
+        Mail mail = FakeMail.builder()
+            .mimeMessage(message)
+            .sender(new MailAddress(FROM))
+            .recipient(new MailAddress(GROUP1_ADDRESS))
+            .build();
+
+        try (SMTPMessageSender messageSender = SMTPMessageSender.noAuthentication(LOCALHOST_IP, SMTP_PORT, DOMAIN1);
+             IMAPMessageReader imapMessageReader = new IMAPMessageReader(LOCALHOST_IP, IMAP_PORT)) {
+            messageSender.sendMessage(mail);
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(messageSender::messageHasBeenSent);
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(() -> imapMessageReader.userReceivedMessage(USER_DOMAIN1, PASSWORD));
+
+            String processedMessage = imapMessageReader.readFirstMessageInInbox(USER_DOMAIN1, PASSWORD);
+
+            assertThat(processedMessage).contains(MESSAGE_CONTENT);
+        }
+
+    }
+
+    @Test
+    public void messageShouldRedirectToUserDoesNotHaveSameDomainWhenBelongingToGroup() throws Exception {
+        RestAssured.given()
+            .port(webAdminGuiceProbe.getWebAdminPort())
+        .when()
+            .put(GroupsRoutes.ROOT_PATH + "/" + GROUP1_ADDRESS + "/" + USER_DOMAIN2)
+        .then()
+            .statusCode(201);
+
+        Mail mail = FakeMail.builder()
+            .mimeMessage(message)
+            .sender(new MailAddress(FROM))
+            .recipient(new MailAddress(GROUP1_ADDRESS))
+            .build();
+
+        try (SMTPMessageSender messageSender = SMTPMessageSender.noAuthentication(LOCALHOST_IP, SMTP_PORT, DOMAIN1);
+             IMAPMessageReader imapMessageReader = new IMAPMessageReader(LOCALHOST_IP, IMAP_PORT)) {
+            messageSender.sendMessage(mail);
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(messageSender::messageHasBeenSent);
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(() -> imapMessageReader.userReceivedMessage(USER_DOMAIN2, PASSWORD));
+
+            String processedMessage = imapMessageReader.readFirstMessageInInbox(USER_DOMAIN2, PASSWORD);
+
+            assertThat(processedMessage).contains(MESSAGE_CONTENT);
+        }
+
+    }
+
+    @Test
+    public void messageShouldRedirectToAllUsersBelongToGroup() throws Exception {
+        RestAssured.given()
```

Comment:
the port should be configured gobally on @Before method.

No need of **then**: we are not testing webadmin CRUD here.

```java
with()
   .put(/* payload */);
```

Should be enough.

Here and everywhere.